fpsim/defaults.py

```diff
@@ -13,61 +13,117 @@
 max_age_preg   = 50   # Maximum age to become pregnant
 max_parity     = 20   # Maximum number of children
 
+
 #%% Defaults when creating a new person
-person_defaults = dict(
-    uid                  = -1,
-    age                  = 0,
-    age_by_group         = 0,
-    sex                  = 0,
-    parity               = 0,
-    method               = 0,
-    barrier              = 0,
-    postpartum_dur       = 0,
-    gestation            = 0,
-    preg_dur             = 0,
-    stillbirth           = 0,
-    miscarriage          = 0,
-    abortion             = 0,
-    short_interval       = 0,
-    secondary_birth      = 0,
-    pregnancies          = 0,
-    months_inactive      = 0,
-    remainder_months     = 0,
-    breastfeed_dur       = 0,
-    breastfeed_dur_total = 0,
-    alive                = True,
-    pregnant             = False,
-    sexually_active      = False,
-    sexual_debut         = False,
-    sexual_debut_age     = -1,
-    first_birth_age      = -1,
-    lactating            = False,
-    postpartum           = False,
-    lam                  = False,
-    mothers              = -1,
-    paid_employment      = False,
-    partnered            = False,
-    partnership_age      = -1,
-    urban                = True,
-    decision_wages       = 0,
-    decision_health      = 0,
-    sexual_autonomy      = 0,
-    edu_objective        = 0,
-    edu_attainment       = 0,
-    edu_dropout          = False,
-    edu_interrupted      = False,
-    edu_completed        = False,
-    edu_started          = False,
+class State:
+    def __init__(self, name, val=None, dtype=None):
+        """
+        Initialize a state
+        Args:
+            name (str): name of state
+            val (list, array, float, or str): value(s) to populate array with
+            dtype (dtype): datatype. Inferred from val if not provided.
+        """
+        self.name = name
+        self.val = val
+        self.dtype = dtype
+
+    def new(self, n, vals=None):
+        """
+        Define an empty array with the correct value and data type
+        """
+        if vals is None: vals = self.val  # Use default if none provided
+
+        if isinstance(vals, np.ndarray):
+            assert len(vals) == n
+            arr = vals
+        elif isinstance(vals, list):
+            arr = [[] for _ in range(n)]
+        else:
+            if self.dtype is None: dtype = object if isinstance(vals, str) else None
+            else: dtype = self.dtype
+            arr = np.full(shape=n, fill_value=vals, dtype=dtype)
+        return arr
+
+
+person_defaults = sc.autolist(
```

Comment:
What's the advantage of this being an `autolist` instead of a regular list? Also feel than an `ndict` could be useful here...